### ISSUE TITLE
fatal error: bulkBarrierPreWrite: unaligned arguments (go 1.19.1)

### ISSUE THREAD

**Issue by akovalenko** [lightningnetwork/lnd#6923]
### Background

I'm running a rather active node on LND, regularly upgrading to git master. I've recently switched from go 1.18.6 to go1.19.1, after which I started receiving a panic (seen two times, each time after several hours of normal work). Both instances of a panic have a backtrace pointing to the same origin (lnwire.go:477). So I went back to 1.18.6, and had 24h+ uptime since then with no panic. 

I'm almost sure that it's a regression in Go, as the LND code in that place seems not doing anything unsafe or unusual. I'm posting this here (1) so you avoid moving to 1.19 for a while until it's resolved, (2) hoping that you'll manage to reproduce this somehow, and  maybe find a more useful reproducer then our current "here's a huge project, it fails each several hours under real-life load"

### Your environment

* version of `lnd`: git 202fcd6672a004881925710e6dc9e336af8945aa
* `Linux 5.15.0-46-generic #49~20.04.1-Ubuntu SMP Thu Aug 4 19:15:44 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
* bitcoind 23.0

### Steps to reproduce

Compile LND with go1.19.1, run a node with (some? big?) traffic for several hours

### Expected behaviour

No panic

### Actual behaviour

Panic

```
Sep 15 17:50:53 murphy lnd[2756008]: fatal error: bulkBarrierPreWrite: unaligned arguments
Sep 15 17:50:53 murphy lnd[2756008]: goroutine 10179 [running]:
Sep 15 17:50:53 murphy lnd[2756008]: runtime.throw({0x1c20812?, 0x9ba5f0?})
Sep 15 17:50:53 murphy lnd[2756008]:         runtime/panic.go:1047 +0x5d fp=0xc0024c2930 sp=0xc0024c2900 pc=0x43807d
Sep 15 17:50:53 murphy lnd[2756008]: runtime.bulkBarrierPreWrite(0x0?, 0x40db47?, 0x20?)
Sep 15 17:50:53 murphy lnd[2756008]:         runtime/mbitmap.go:582 +0x428 fp=0xc0024c29a0 sp=0xc0024c2930 pc=0x415068
Sep 15 17:50:53 murphy lnd[2756008]: runtime.typedmemmove(0x1867fc0, 0xc0083db3f1, 0xc0024c2a84)
Sep 15 17:50:53 murphy lnd[2756008]:         runtime/mbarrier.go:162 +0x45 fp=0xc0024c29d8 sp=0xc0024c29a0 pc=0x414245
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/lnwire.ReadElement({0x1f7b820, 0xc01151c7b0}, {0x18b19a0?, 0xc0083db3f1?})
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/lnwire/lnwire.go:477 +0x173c fp=0xc0024c2cc8 sp=0xc0024c29d8 pc=0x9b8b5c
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/lnwire.ReadElements(...)
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/lnwire/lnwire.go:879
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/lnwire.(*NodeAnnouncement).Decode(0x1f7b820?, {0x1f7b820, 0xc01151c7b0}, 0x2?)
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/lnwire/node_announcement.go:112 +0x150 fp=0xc0024c2d88 sp=0xc0024c2cc8 pc=0x9ba8b0
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/lnwire.ReadMessage({0x1f7b820, 0xc01151c7b0}, 0x1c8?)
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/lnwire/message.go:314 +0x91 fp=0xc0024c2de8 sp=0xc0024c2d88 pc=0x9ba1b1
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/peer.(*Brontide).readNextMessage.func1(0xc003d3c000)
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/peer/brontide.go:1114 +0x146 fp=0xc0024c2e60 sp=0xc0024c2de8 pc=0x164e026
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/pool.(*Read).Submit.func1({0x1f87e28?, 0xc002680ca0?})
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/pool/read.go:50 +0x37 fp=0xc0024c2e88 sp=0xc0024c2e60 pc=0x1648557
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/pool.(*Worker).spawnWorker(0xc0088eae60, 0xc002680c90)
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/pool/worker.go:232 +0x316 fp=0xc0024c2fc0 sp=0xc0024c2e88 pc=0x1649556
Sep 15 17:50:53 murphy lnd[2756008]: github.com/lightningnetwork/lnd/pool.(*Worker).requestHandler.func2()
Sep 15 17:50:53 murphy lnd[2756008]:         github.com/lightningnetwork/lnd/pool/worker.go:161 +0x2a fp=0xc0024c2fe0 sp=0xc0024c2fc0 pc=0x16491aa
Sep 15 17:50:53 murphy lnd[2756008]: runtime.goexit()
Sep 15 17:50:53 murphy lnd[2756008]:         runtime/asm_amd64.s:1594 +0x1 fp=0xc0024c2fe8 sp=0xc0024c2fe0 pc=0x46c241

```

**Comment by Roasbeef** [comment]
This looks related tho the issue OP reports the issue on iOS: https://github.com/golang/go/issues/46893

**Comment by akovalenko** [comment]
I build with CGO_ENABLED=0, so looks unrelated to me

**Comment by guggero** [comment]
I was able to reproduce quite easily with go 1.19.1 in an itest.
Looks like `*e = alias` triggers this in [`ReadElement`](https://github.com/lightningnetwork/lnd/blob/master/lnwire/lnwire.go#L477).

There's now an issue open that sounds like exactly our case (`NodeAlias` is a `[32]byte` type): https://github.com/golang/go/issues/55124

**Comment by Roasbeef** [comment]
Ok the fix in Go is now being tracked here: https://github.com/golang/go/issues/55124

**Comment by alexbosworth** [comment]
Fixed now in Go 1.19.2?

**Comment by akovalenko** [comment]
@alexbosworth yes (I've checked asm output to be sure)
